urldd9e4ada6f_kirillF_sample_jenkins_project_tgz-pJ8-com_bigfatplayer_hello_AppJ8/src/com/bigfatplayer/hello/Work.java

```diff
@@ -0,0 +1,583 @@
+    this.comment = comment;
+    return this;
+  }
+
+  public void unsetComment() {
+    this.comment = null;
+  }
+
+  /** Returns true if field comment is set (has been assigned a value) and false otherwise */
+  public boolean isSetComment() {
+    return this.comment != null;
+  }
+
+  public void setCommentIsSet(boolean value) {
+    if (!value) {
+      this.comment = null;
+    }
+  }
+
+  public void setFieldValue(_Fields field, Object value) {
+    switch (field) {
+    case NUM1:
+      if (value == null) {
+        unsetNum1();
+      } else {
+        setNum1((Integer)value);
+      }
+      break;
+
+    case NUM2:
+      if (value == null) {
+        unsetNum2();
+      } else {
+        setNum2((Integer)value);
+      }
+      break;
+
+    case COMMENT:
+      if (value == null) {
+        unsetComment();
+      } else {
+        setComment((String)value);
+      }
+      break;
+
+    }
+  }
+
+  public Object getFieldValue(_Fields field) {
+    switch (field) {
+    case NUM1:
+      return Integer.valueOf(getNum1());
+
+    case NUM2:
+      return Integer.valueOf(getNum2());
+
+    case COMMENT:
+      return getComment();
+
+    }
+    throw new IllegalStateException();
+  }
+
+  /** Returns true if field corresponding to fieldID is set (has been assigned a value) and false otherwise */
+  public boolean isSet(_Fields field) {
+    if (field == null) {
+      throw new IllegalArgumentException();
+    }
+
+    switch (field) {
+    case NUM1:
+      return isSetNum1();
+    case NUM2:
+      return isSetNum2();
+    case COMMENT:
+      return isSetComment();
+    }
+    throw new IllegalStateException();
+  }
+
+  @Override
+  public boolean equals(Object that) {
+    if (that == null)
+      return false;
+    if (that instanceof Work)
+      return this.equals((Work)that);
+    return false;
+  }
+
+  public boolean equals(Work that) {
+    if (that == null)
+      return false;
+
+    boolean this_present_num1 = true;
+    boolean that_present_num1 = true;
+    if (this_present_num1 || that_present_num1) {
+      if (!(this_present_num1 && that_present_num1))
+        return false;
+      if (this.num1 != that.num1)
+        return false;
+    }
+
+    boolean this_present_num2 = true;
+    boolean that_present_num2 = true;
+    if (this_present_num2 || that_present_num2) {
+      if (!(this_present_num2 && that_present_num2))
+        return false;
+      if (this.num2 != that.num2)
+        return false;
+    }
+
+    boolean this_present_comment = true && this.isSetComment();
+    boolean that_present_comment = true && that.isSetComment();
+    if (this_present_comment || that_present_comment) {
+      if (!(this_present_comment && that_present_comment))
+        return false;
+      if (!this.comment.equals(that.comment))
+        return false;
+    }
+
+    return true;
+  }
+
+  @Override
+  public int hashCode() {
+    return 0;
+  }
+
+  public int compareTo(Work other) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 11. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.